src/test/java/com/crud/tasks/controller/TrelloControllerTest.java

```diff
@@ -0,0 +1,110 @@
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private TrelloFacade trelloFacade;
+
+    @MockBean
+    private TrelloBadgeDto trelloBadgeDto;
+
+    @Test
+    public void shouldFetchEmptyTrelloBoards() throws Exception {
+        // Given
+        List<TrelloBoardDto> trelloBoards = new ArrayList<>();
+        when(trelloFacade.fetchTrelloBoards()).thenReturn(trelloBoards);
+
+        // When & Then
+        mockMvc.perform(get("/v1/trello/getTrelloBoards").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().is(200)) // or isOk((
+                .andExpect(jsonPath("$", hasSize(0)));
+    }
+
+    @Test
+    public void shouldFetchTrelloBoards() throws Exception {
+        // Given
+        List<TrelloListDto> trelloLists = new ArrayList<>();
+        trelloLists.add(new TrelloListDto("1", "Test list", false));
+
+        List<TrelloBoardDto> trelloBoards = new ArrayList<>();
+        trelloBoards.add(new TrelloBoardDto("1", "Test task", trelloLists));
+
+        when(trelloFacade.fetchTrelloBoards()).thenReturn(trelloBoards);
+
+        // When & Then
+
+        mockMvc.perform(get("/v1/trello/getTrelloBoards").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                //Trello board fields
+                .andExpect(jsonPath("$", hasSize(1)))
+                .andExpect(jsonPath("$[0].id", is("1")))
+                .andExpect(jsonPath("$[0].name", is("Test task")))
+                //Trello list fields
+                .andExpect(jsonPath("$[0].lists", hasSize(1)))
+                .andExpect(jsonPath("$[0].lists[0].id", is("1")))
+                .andExpect(jsonPath("$[0].lists[0].name", is("Test list")))
+                .andExpect(jsonPath("$[0].lists[0].closed", is(false)));
+    }
+
+    @Test
+    public void shouldCreateTrelloCard() throws Exception {
+        // Given
+        TrelloCardDto trelloCardDto = new TrelloCardDto(
+                "Test",
+                "Test description",
+                "top",
+                "1");
+
+        CreatedTrelloCardDto createdTrelloCardDto = new CreatedTrelloCardDto(
+                "323",
+                trelloBadgeDto,
+                "Test",
+                "http://test.com");
+
+        when(trelloFacade.createCard(ArgumentMatchers.any(TrelloCardDto.class))).thenReturn(createdTrelloCardDto);
+
+        Gson gson = new Gson();
+        String jsonContent = gson.toJson(trelloCardDto);
+
+        // When & Then
+        mockMvc.perform(post("/v1/trello/createTrelloCard")
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding("UTF-8")
+                .content(jsonContent))
+                .andExpect(jsonPath("$.id", is("323")))
+                .andExpect(jsonPath("$.trelloBadgeDto", trelloBadgeDto))
+                .andExpect(jsonPath("$.name", is("Test")))
+                .andExpect(jsonPath("$.shortUrl", is("http://test.com")));
+    }
```

Comment:
createTrelloCard zwraca obiekt 
```

@AllArgsConstructor
@Getter
@JsonIgnoreProperties(ignoreUnknown = true)
public class CreatedTrelloCardDto {
    @JsonProperty("id")
    private String id;

    @JsonProperty("badges")
    private TrelloBadgeDto trelloBadgeDto;

    @JsonProperty("name")
    private String name;

    @JsonProperty("shortUrl")
    private String shortUrl;
}
```

Reply:
wiec musisz sprawdzic pola 
* id
* trelloBadgeDto
* name
* shortUrl

a nie TrelloBadgeDto

Reply:
gdybys gdzies potrzebowal przetestowac TrelloBadgeDto, to potrzebujesz wchodzic w głąb json'a w stylu 
`$.attachmentsByType.trello.board`